src/lib.rs

```diff
@@ -201,6 +203,39 @@ pub enum IngestError {
         /// The version returned after the commit
         actual_version: DeltaDataTypeVersion,
     },
+    /// Error returned if unable to construct a deserializer
+    #[error("Unable to construct a message deserializer, source: {source}")]
+    UnableToCreateDeserializer {
+        /// The underlying error.
+        source: anyhow::Error,
+    },
+}
+
+/// Formats for message parsing
+#[derive(Clone)]
+pub enum MessageFormat {
+    /// Parses messages as json and uses the inferred schema
+    DefaultJson,
+
+    /// Parses messages as json using the provided schema source. Will not use json schema files
+    Json(SchemaSource),
+
+    /// Parses avro messages using provided schema, schema registry or schema within file
+    Avro(SchemaSource),
+    //    ProtoBuf(SchemaSource),
+}
+
+/// Source for schema
+#[derive(Clone)]
+pub enum SchemaSource {
+    /// Use default behavior
+    None,
+
+    /// Use confluent schema registry url
+    SchemaRegistry(String),
```

Comment:
```suggestion
    SchemaRegistry(Url),
```

We already depend on the `url` crate to where I think it would be worth strongly typing this